src/hardware/mixer.cpp

```diff
@@ -117,33 +121,71 @@ static struct mixer_t mixer = {};
 
 uint8_t MixTemp[MIXER_BUFSIZE] = {};
 
-MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name)
+MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name,
+                           const std::set<ChannelFeature> &_features)
         : name(_name),
           envelope(_name),
-          handler(_handler)
+          handler(_handler),
+          features(_features)
 {}
 
+bool MixerChannel::HasFeature(ChannelFeature feature)
```

Comment:
Read-only argument, and the member function itself won't update the object, so can be `const` as well.